### PR TITLE
RHICOMPL-199 - Cache failed rules results

### PR DESCRIPTION
**Pull request by bastilian**

### CODE REVIEW

File: app/models/schema.rb [hunk 1]
```diff
@@ -127,12 +127,14 @@
     type types[RuleType]
     description 'Rules failed by a system'
     resolve lambda { |host, _args, _ctx|
-      Rule.where(
-        id: RuleResult.includes(:rule).where(
-          host: host,
-          result: %w[error fail notchecked]
-        ).pluck(:rule_id).uniq
-      )
+      Rails.cache.fetch("#{host.id}/rule_objects_result", expires_in: 1.week) do
```
Comment (dLobatog) [inline app/models/schema.rb]: Looks good - can you invalidate the cache when a new report comes in? We normally do that in https://github.com/RedHatInsights/compliance-backend/blob/master/app/services/xccdf_report_parser.rb#L72 - this looks like just 1 operation, no need to put it on a loop like that one

Comment (bastilian) [reply app/models/schema.rb]: Thanks, that was what I was looking for!

**Comment by bastilian** [comment]
@dLobatog Added the cache deletion and renamed it to reflect what it contains better.

File: app/services/xccdf_report_parser.rb [hunk 2]
```diff
@@ -71,6 +71,7 @@ def save_all
       rules_already_saved.each do |rule|
         Rails.cache.delete("#{rule.id}/#{@host.id}/compliant")
       end
+      Rails.cache.delete("#{host.id}/failed_rule_objects_result")
```
Comment (dLobatog) [inline app/services/xccdf_report_parser.rb]: `@host` instead of host

**Pull request merged**